tests/nn/data_parallel/test_fsdp_regnet.py

```diff
@@ -9,86 +9,216 @@
 
 """ Test FSDP with regnet-like model. """
 
+import contextlib
 import random
 import tempfile
 
 import pytest
 import torch
 import torch.multiprocessing as mp
 from torch.nn import BatchNorm2d, Conv2d, Module, SyncBatchNorm
+from torch.nn.parallel import DistributedDataParallel as DDP
 from torch.optim import SGD
 
 from fairscale.nn.data_parallel import FullyShardedDataParallel as FSDP
 from fairscale.nn.data_parallel import TrainingState, auto_wrap_bn
-from fairscale.utils.testing import dist_init, skip_if_single_gpu, teardown, torch_version
+from fairscale.utils.testing import (
+    dist_init,
+    objects_are_equal,
+    rmf,
+    skip_if_single_gpu,
+    state_dict_norm,
+    teardown,
+    torch_version,
+)
+
+
+class Model(Module):
+    def __init__(self):
+        super().__init__()
+        # TODO (Min): for now, we just test pytorch sync_bn here.
+        #             this will grow into regnet; testing apex sync_bn, etc.
+        self.conv = Conv2d(2, 2, (1, 1))
+        self.bn = BatchNorm2d(2)
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.bn(x)
+        return x
+
+
+# We get a bit fancy here. Since the scope is `module`, this is run only
```

Comment:
nice, I did not know this, thanks for the explanations

Reply:
I was learning it the first time too.